2015-02-27  Marek Polacek  <polacek@redhat.com>

	PR c/65040
	* doc/invoke.texi: Update to reflect that -Wformat=2 doesn't enable
	-Wformat-signedness anymore.
c-family/
	* c.opt (Wformat-signedness): Don't enable by -Wformat=2.
testsuite/
	* gcc.dg/pr65066.c: Use -Wformat -Wformat-signedness and not
	-Wformat=2.

--- gcc/doc/invoke.texi	(revision 221060)
+++ gcc/doc/invoke.texi	(revision 221061)
@@ -3631,7 +3631,7 @@ The C standard specifies that zero-lengt
 @opindex Wformat=2
 Enable @option{-Wformat} plus additional format checks.  Currently
 equivalent to @option{-Wformat -Wformat-nonliteral -Wformat-security
--Wformat-signedness -Wformat-y2k}.
+-Wformat-y2k}.
 
 @item -Wformat-nonliteral
 @opindex Wformat-nonliteral
--- gcc/c-family/c.opt	(revision 221060)
+++ gcc/c-family/c.opt	(revision 221061)
@@ -456,7 +456,7 @@ C ObjC C++ ObjC++ Var(warn_format_securi
 Warn about possible security problems with format functions
 
 Wformat-signedness
-C ObjC C++ ObjC++ Var(warn_format_signedness) Warning LangEnabledBy(C ObjC C++ ObjC++,Wformat=, warn_format >= 2, 0)
+C ObjC C++ ObjC++ Var(warn_format_signedness) Warning
 Warn about sign differences with format functions
 
 Wformat-y2k
--- gcc/testsuite/gcc.dg/pr65066.c	(revision 221060)
+++ gcc/testsuite/gcc.dg/pr65066.c	(revision 221061)
@@ -1,6 +1,6 @@
 /* PR c/65066 */
 /* { dg-do compile } */
-/* { dg-options "-Wformat=2" } */
+/* { dg-options "-Wformat -Wformat-signedness" } */
 
 extern int sscanf (const char *restrict, const char *restrict, ...);
 int *a;
